packages/backend/src/matter/devices/media-player-device.ts

```diff
@@ -1,17 +1,51 @@
-import { MatterDevice } from "../matter-device.js";
-import { OnOffPlugInUnitDevice } from "@matter/main/devices";
-import { OnOffServer } from "../behaviors/on-off-server.js";
+import {
+  HomeAssistantEntityState,
+  SpeakerDeviceAttributes,
+} from "@home-assistant-matter-hub/common";
+import { SpeakerDevice } from "@matter/main/devices";
 import { BasicInformationServer } from "../behaviors/basic-information-server.js";
 import { IdentifyServer } from "../behaviors/identify-server.js";
 import { HomeAssistantBehavior } from "../custom-behaviors/home-assistant-behavior.js";
+import { OnOffServer } from "../behaviors/on-off-server.js";
+import { MatterDevice } from "../matter-device.js";
+import {
+  LevelControlConfig,
+  LevelControlServer,
+} from "../behaviors/level-control-server.js";
+import { SwitchDevice } from "./switch-device.js";
+
+const volumeLevelConfig: LevelControlConfig = {
+  getValue: (state: HomeAssistantEntityState<SpeakerDeviceAttributes>) => {
+    if (state.attributes.volume_level != null) {
+      return state.attributes.volume_level * 254;
+    }
+    return 0;
+  },
+  moveToLevel: {
+    action: "media_player.volume_set",
+    data: (value) => {
+      return { volume_level: value / 100 };
+    },
+  },
+};
 
-const MediaPlayerEndpointType = OnOffPlugInUnitDevice.with(
+const MediaPlayerEndpointType = SpeakerDevice.with(
```

Comment:
Awesome! I didn’t know that this is already supported  by Google Home. I’ll test it tomorrow with Alexa before merging. 

Unfortunately Alexa drops the whole bridge if one unsupported entity is included.

Does speaker device supports any media streaming? Does Google Home support it?

Reply:
Seems it only supports On/Off and level control: https://developers.home.google.com/matter/supported-devices#:~:text=Water%20Pump-,Speaker,-Level%20Control%0AOn 

Maybe we could override the mapping  of `media_player` via config?

Reply:
Yes, I want to introduce configs per entity and per domain anyway on bridge level. So one can configure it per bridge depending on what they want to pair with it. 

But I have no concept yet how to integrate it with all device types. 

Let me test it with Alexa first.

Reply:
Okay, it doesn't work with Alexa 😢 
So we need to introduce some option to configure it...

Reply:
I have added three options to configure devices:
- per domain
- per entity_id
- compatibility mode on bridge-level

The first two just add the possibility to add configs later. There is NO config at all existing yet.
Anyway in your case here, i'd suggest going with compatibility mode.

For `matter` and `GoogleHome` use the new SpeakerDevice Type.
For the rest use the "old" SwitchType.

I've already added the compatibility as unused parameter in `MediaPlayerDevice`-